testsuite/unittests/test_inh_stdp_synapse.sli

```diff
@@ -0,0 +1,245 @@
+/post_spikes sd_post GetStatus /events get /times get cva { backpr_delay add } Map def
+
+
+/final_weight << /source parrot /target neuron >> FindConnections 0 get /weight get def
+
+%%% check final weight %%%
+
+
+cout 15 setprecision
+
+/K_plus    0.0 def
+/K_minus   0.0 def
+/last_pre  0   def
+/last_post 0   def
+/j         0   def
+/i         0   def
+
+/post_spike post_spikes i get def
+/pre_spike  pre_spikes  j get def
+/w          w_neuron w_max div   def
+
+/update_K_plus
+{
+  last_pre pre_spike sub tau_plus div exp K_plus mul 1.0 add /K_plus Set
+}
+def
+
+/update_K_minus
+{
+  last_post post_spike sub tau_minus div exp K_minus mul 1.0 add /K_minus Set
+}
+def
+
+/next_pre_spike
+{
+  j 1 add /j Set
+  pre_spike /last_pre Set
+  pre_spikes j get /pre_spike Set
+}
+def
+
+/next_post_spike
+{
+  i 1 add /i Set
+  post_spike /last_post Set
+  post_spikes i get /post_spike Set
+}
+def
+
+/facilitate
+{
+  ( w + lambda * (1.0-w)**mu_plus * K_plus * exp((last_pre-post_spike)/tau_plus) ) ExecMath
+  dup 1.0 lt { /w Set } { pop 1.0 /w Set } ifelse
+  %(facilitation) =only (\t) =only last_pre =only (\t) =only post_spike =only (\t) =only w w_max mul =
+}
+def
+
+/depress
+{
+  ( w - lambda * alpha * w**mu_minus * K_minus * exp((last_post-pre_spike)/tau_minus) ) ExecMath
+  dup 0.0 gt { /w Set } { pop 0.0 /w Set } ifelse
+  %(depression) =only (\t) =only last_post =only (\t) =only pre_spike =only (\t) =only w w_max mul =
+}
+def
+
+{
+  pre_spike post_spike eq
+  { % pre- and post-syn. spike at the same time
+    last_post post_spike neq { facilitate } if
+    last_pre pre_spike neq { depress } if
+    %(pre == post) =only (\t) =only pre_spike =only (\t) =only post_spike =only (\t) =only w w_max mul =
+    j 1 add pre_spikes length lt
+    {
+      update_K_plus
+      next_pre_spike
+      i 1 add post_spikes length lt
+      {
+        update_K_minus
+	next_post_spike
+      }
+      if
+    }
+    {
+      exit
+    }
+    ifelse
+  }
+  {
+    pre_spike post_spike lt
+    { % next spike is a pre-syn. spike
+      depress
+      update_K_plus
+      j 1 add pre_spikes length lt
+      {
+        next_pre_spike
+      }
+      {
+        %(last presyn spike) =
+        % we don't consider the post-syn. spikes after the last pre-syn. spike
+        exit
+      }
+      ifelse
+    }
+    { % next spike is a post-syn. spike
+      facilitate
+      update_K_minus
+      i 1 add post_spikes length lt
+      {
+        next_post_spike
+      }
+      {
+        %(last postsyn spike) =
+        % we DO consider the pre-syn. spikes after the last post-syn. spike
+        post_spike /last_post Set
+        pre_spikes dup length 1 sub get resolution add /post_spike Set  % to make sure we don't come here again
+      }
+      ifelse
+    }
+    ifelse
+  }
+  ifelse
+}
+loop
+
+%final_weight w w_max mul sub abs dup = 1e-13 leq =
+final_weight w w_max mul sub abs 1e-13 leq assert_or_die
```

Comment:
It would be more in keeping the testsuite style to put the entire loop from l 185 on up to and including the `leq` into a block and the place the `assert_or_die` after the block. Also, there is a SLI function `ToUnittestPrecision` (or similar) that allows testing with a prescribed number of decimals.